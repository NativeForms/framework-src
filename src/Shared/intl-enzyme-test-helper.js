/**
 * Components using the react-intl module require access to the intl context.
 * This is not available when mounting single components in Enzyme.
 * These helper functions aim to address that and wrap a valid,
 * English-locale intl context around them.
 */

import React from 'react';
import { IntlProvider, intlShape } from 'react-intl';
import { mount, shallow } from 'enzyme'; // eslint-disable-line import/no-extraneous-dependencies

/** Create the IntlProvider to retrieve context for wrapping around. */
function createIntlContext(messages, locale) {
  const intlProvider = new IntlProvider({ messages, locale }, {});
  const { intl } = intlProvider.getChildContext();
  return intl;
}

/** When using React-Intl `injectIntl` on components, props.intl is required. */
function nodeWithIntlProp(node, messages = {}, locale = 'en') {
  return React.cloneElement(node, { intl: createIntlContext(messages, locale) });
}

/**
 * Create a shadow renderer that wraps a node with Intl provider context.
 * @param {ReactComponent} node - Any React Component
 * @param {Object} context
 * @param {Object} messages - A map with keys (id) and messages (value)
 * @param {string} locale - Locale string
 */
export function shallowWithIntl(node, { context } = {}, messages = {}, locale = 'en') {
  return shallow(
    nodeWithIntlProp(node),
    {
      context: Object.assign({}, context, { intl: createIntlContext(messages, locale) }),
    }
  );
}

/**
 * Mount the node with Intl provider context.
 * @param {Component} node - Any React Component
 * @param {Object} context
 * @param {Object} messages - A map with keys (id) and messages (value)
 * @param {string} locale - Locale string
 */
export function mountWithIntl(node, { context, childContextTypes } = {}, messages = {}, locale = 'en') {
  return mount(
    nodeWithIntlProp(node),
    {
      context: Object.assign({}, context, { intl: createIntlContext(messages, locale) }),
      childContextTypes: Object.assign({}, { intl: intlShape }, childContextTypes)
    }
  );
}
