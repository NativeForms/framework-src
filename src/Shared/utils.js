import React from 'react';
import syntaxHighlighter from 'react-native-syntax-highlighter';
import { github } from 'react-syntax-highlighter/dist/styles';
import { Iterable } from 'immutable';

export const SchemaViewer = (props) => syntaxHighlighter(
  Object.assign({}, syntaxHighlighter.defaultProps, {
    language: 'javascript',
    style: github,
    codeTagProps: {
      alwaysBounceVertical: false
    }
  }, props));

// See http://redux.js.org/docs/recipes/UsingImmutableJS.html#use-a-higher-order-component-to-convert-your-smart-components-immutablejs-props-to-your-dumb-components-javascript-props
export const toJS = (WrappedComponent) =>
  (wrappedComponentProps) => {
    const KEY = 0;
    const VALUE = 1;

    const propsJS = Object.entries(wrappedComponentProps)
        .reduce((newProps, wrappedComponentProp) => {
          /* eslint no-param-reassign:0 */
          newProps[wrappedComponentProp[KEY]] =
              Iterable.isIterable(wrappedComponentProp[VALUE])
                  ? wrappedComponentProp[VALUE].toJS()
                  : wrappedComponentProp[VALUE];
          return newProps;
        }, {});

    return <WrappedComponent {...propsJS} />;
  };

export function getLocale() {
  return 'en';
}
