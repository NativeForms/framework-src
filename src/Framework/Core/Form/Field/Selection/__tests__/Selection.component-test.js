import React from 'react';

import SelectionComponent from '../Selection.component';
import i18n from '../Selection.i18n';
import schema from '../Selection.schema';

//shared
import { mountWithIntl } from '../../../../../../Shared/intl-enzyme-test-helper';

const message = i18n.en;
const propsArray = schema.attributes.formComponents;

describe('components <SelectionComponent />', () => {
  let wrapper;
  propsArray.forEach((props) => {
    const { attributes, ...rest } = props;
    it('renders text input component correctly', () => {
      wrapper = mountWithIntl(<SelectionComponent {...rest} />, {}, message);
      expect(wrapper.find('SelectionComponent')).toMatchSnapshot();
    });
  });
});
