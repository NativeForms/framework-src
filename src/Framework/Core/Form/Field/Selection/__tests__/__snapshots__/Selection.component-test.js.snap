// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components <SelectionComponent /> renders text input component correctly 1`] = `
<SelectionComponent
  category="field"
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": Object {
        "label": "Selection Label",
        "title": "Selection",
      },
      "now": [Function],
      "textComponent": "span",
    }
  }
  label={null}
  radioBoxType={false}
  selectionList={Array []}
  type="selection"
  uid="noHeaderSelection"
>
  <Styled(List)>
    <List
      style={Object {}}
    >
      <View
        style={Object {}}
      >
        <View
          style={Object {}}
        />
      </View>
    </List>
  </Styled(List)>
</SelectionComponent>
`;

exports[`components <SelectionComponent /> renders text input component correctly 2`] = `
<SelectionComponent
  category="field"
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": Object {
        "label": "Selection Label",
        "title": "Selection",
      },
      "now": [Function],
      "textComponent": "span",
    }
  }
  label={null}
  radioBoxType={false}
  selectionList={Array []}
  type="selection"
  uid="selectionExample"
>
  <Styled(List)>
    <List
      style={Object {}}
    >
      <View
        style={Object {}}
      >
        <View
          style={Object {}}
        />
      </View>
    </List>
  </Styled(List)>
</SelectionComponent>
`;

exports[`components <SelectionComponent /> renders text input component correctly 3`] = `
<SelectionComponent
  category="field"
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": Object {
        "label": "Selection Label",
        "title": "Selection",
      },
      "now": [Function],
      "textComponent": "span",
    }
  }
  label={null}
  radioBoxType={false}
  selectionList={Array []}
  type="selection"
  uid="radioSelectionExample"
>
  <Styled(List)>
    <List
      style={Object {}}
    >
      <View
        style={Object {}}
      >
        <View
          style={Object {}}
        />
      </View>
    </List>
  </Styled(List)>
</SelectionComponent>
`;
